hal/riscv64/cpu.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * RV64 CPU related routines
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _CPU_H_
+#define _CPU_H_
+
+
+/* CSR bits */
+#define SSTATUS_SIE (1 << 1)
+#define SSTATUS_FS  (3 << 13)
+
+#define SCAUSE_IRQ_MSK (0xff)
+#define SCAUSE_INTR    (1 << 63)
+
+/* Interrupts */
+#define CLINT_IRQ_FLG (1u << 31) /* Marks that interrupt handler is installed for CLINT, not PLIC */
+
+#ifndef __ASSEMBLY__
+
+
+#define csr_set(csr, val) \
+	({ \
+		unsigned long __v = (unsigned long)(val); \
+		__asm__ __volatile__("csrs " #csr ", %0" \
+							 : \
+							 : "rK"(__v) \
+							 : "memory"); \
```

Comment:
clang-format is getting confused by inline asm
```suggestion
		__asm__ __volatile__("csrs " #csr ", %0" :: "rK"(__v) : "memory");
```